mlir/include/mlir/Dialect/MIOpenOps/LowerMIOpenOps.h

```diff
@@ -149,6 +153,11 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
         layoutAttr0.append(sourceNonKDimAttr.begin(), sourceNonKDimAttr.end());
         layoutAttr1.append(targetNonKDimAttr.begin(), targetNonKDimAttr.end());
         layoutAttr1.append(sourceKDimAttr.begin(), sourceKDimAttr.end());
+      } else if (convOpType == miopen::ConvOpType::Conv2DBwdWeightOpType) {
+        layoutAttr0.append(targetKDimAttr.begin(), targetKDimAttr.end());
+        layoutAttr0.append(sourceKDimAttr.begin(), sourceKDimAttr.end());
+        layoutAttr1.append(targetNonKDimAttr.begin(), targetNonKDimAttr.end());
+        layoutAttr1.append(sourceNonKDimAttr.begin(), sourceNonKDimAttr.end());
```

Comment:
Change to `targetGemmMAttr` and `targetGemmNDimAttr`?

Reply:
You can do that. Basically create two more attributes but only assign them when its bww. Remember to add a comment on it, as it can get very confusing.

Reply:
I think the K here is K in convolution. targetKDimAttr and targetNonKDimAttr actually is good, since for all directions, the transformation will always be CYX and K. The only difference is mapping to GemmM, GemmN, and GemmK.

Reply:
Yes, K is the target gemm* in convolution. The mapping is the confusing part: For example, if makes sense when the filter's act as either gemmK or non-gemmK when the actual convolution is [gemmK, gemmM] or [gemmK, gemmN]. However, the actual convolution in bww is [gemmM, gemmN]. None of them are K. If is very hard for another developer to get gemmK is mapped to gemmM, for example, because it is not explicit.

Reply:
I thought the K here means K in KCYX.

Reply:
Hmm as long as I remember, the `sourceK` means `K` in `KCYX`; `targetK` means `K` in `gemmk gemmM`. Could you double check the context and make sure it make sense.

Honestly with your latest changes that make it possible that K may doesn't exist at all. It is probably a better idea we just get rid of all `K`s in our variable naming. Such that one naming convention can fits to all cases.

How about something like replacing `K` to `LayoutPos1` in source dimensions(short for layout position 1, if it is `kcyx`, then it represents for k.), and replace `K` to `gemmPos1` in target dimensions(short for gemm position 1`, if it is `gemmk gemmM` then it represents gemmk)?

Let me know if you think this is a better idea.

Reply:
In MIOpen, we use ProblemK and GemmK to distinguish. To be consistent, how about for sourceDim, we use sourceProblemKDim and sourceNoProblemKDim to present K and CYX, for target, we use targetGemmMDim and targetNonGemmMDim to present GemmM and GemmK/N?

Reply:
@zjing14 Thanks for the information on `Problem/Gemm` terminology. Though I think the same issue exists here. When looking at a specific gemm position, for example filter's target dimension. It can either be `gemmK gemmM`, `gemmK gemmN`or `gemmM gemmN`. 

What I think that makes more sense is to find something more generic than the specific letter of gemm. When we pick the naming convention to be `gemmK` and non-`gemmK`, it will not fit for when there is not `gemmK` in it in the target dimension. In this context, the proposal I have is something like `gemmPos1`. If we can agree on the problem with the letter based naming, could you come up with something more generic and not letter dependent?

Reply:
As I'm amending my PR, I think the we always define the target gemm in the gridwise gemm to be:
   `gemmA(K,M) x gemmB(K,N) = gemmC(M,N)`

In this way, we can have the following naming scheme:

- Use position in source gemm: `sourceGemmPos0`, `sourceGemmPos1`
- Use gemmA/B/C in target gemm: `gemmADimK`, `gemmADimM`

Reply:
Oppositely, for source, we can keep current naming: sourceKDim, and sourceCYXDim, since currently, I do not see transformations on filter other than passthrough K, and merge CYX. For target, we can use targetGemmPos0 and targetGemmPos1.

Reply:
Sounds good to me. Thanks